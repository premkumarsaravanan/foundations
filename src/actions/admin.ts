import { actionCreator } from '../utils/actions'
import ActionTypes from '../constants/action-types'
import { AppRevisionModel } from '../types/marketplace-api-schema'

export const adminLoading = actionCreator<boolean>(ActionTypes.ADMIN_LOADING)
export const adminReceiveRevisions = actionCreator<AppRevisionModel[] | undefined>(ActionTypes.ADMIN_RECEIVE_REVISIONS)
export const adminRequestRevisions = actionCreator<void>(ActionTypes.ADMIN_REQUEST_REVISIONS)
