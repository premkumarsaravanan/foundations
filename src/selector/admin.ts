import { ReduxState } from '@/types/core'

export const selectAdminAppsState = (state: ReduxState) => {
  return state.adminApps
}

export const selectAdminAppsData = (state: ReduxState) => {
  return state.adminApps.adminAppsData
}
