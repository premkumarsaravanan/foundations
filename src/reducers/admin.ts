import { Action } from '../types/core'
import { isType } from '../utils/actions'
import { adminLoading, adminReceiveRevisions } from '../actions/admin'
import { AppRevisionModel } from '../types/marketplace-api-schema'

export interface AdminState {
  loading: boolean
  appRevisions: AppRevisionModel[] | null
}

export const defaultState: AdminState = {
  loading: false,
  appRevisions: null
}

const adminReducer = (state: AdminState = defaultState, action: Action<any>): AdminState => {
  if (isType(action, adminLoading)) {
    return {
      ...state,
      loading: action.data
    }
  }

  if (isType(action, adminReceiveRevisions)) {
    return {
      ...state,
      loading: false,
      appRevisions: action.data || null
    }
  }

  return state
}

export default adminReducer
