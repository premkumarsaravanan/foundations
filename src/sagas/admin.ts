// import fetcher from '../utils/fetcher'
// import { URLS } from '../constants/api'
import { adminLoading, adminReceiveRevisions } from '../actions/admin'
import { put, delay, fork, takeLatest, all } from '@redux-saga/core/effects'
import ActionTypes from '../constants/action-types'
import { errorThrownServer } from '../actions/error'
import errorMessages from '../constants/error-messages'
import { AppRevisionModel } from '../types/marketplace-api-schema'

export const mockMyAppRevisions: AppRevisionModel[] = []

export const appRevisionsFetch = function*() {
  yield put(adminLoading(true))

  try {
    yield delay(1000)
    yield put(adminReceiveRevisions(mockMyAppRevisions))
  } catch (err) {
    console.error(err.message)
    yield put(
      errorThrownServer({
        type: 'SERVER',
        message: errorMessages.DEFAULT_SERVER_ERROR
      })
    )
  }
}

export const appRevisionsDataListen = function*() {
  yield takeLatest(ActionTypes.ADMIN_REQUEST_REVISIONS, appRevisionsFetch)
}

const adminSagas = function*() {
  yield all([fork(appRevisionsDataListen)])
}

export default adminSagas
