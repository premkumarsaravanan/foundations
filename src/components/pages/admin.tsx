import * as React from 'react'
import { connect } from 'react-redux'
import { ReduxState } from '@/types/core'
import { Loader } from '@reapit/elements'
import ErrorBoundary from '@/components/hocs/error-boundary'
import { AdminState } from '@/reducers/admin'
import bulma from '../../styles/vendor/bulma'

export interface AdminHomeMappedProps {
  adminState: AdminState
}

export type AdminHomeProps = AdminHomeMappedProps

const { container } = bulma

export const AdminHome: React.FunctionComponent<AdminHomeProps> = ({ adminState }) => (
  <div className={container}>
    {adminState.loading ? (
      <Loader />
    ) : (
      <ErrorBoundary>
        <div>Admin area</div>
      </ErrorBoundary>
    )}
  </div>
)

const mapStateToProps = (state: ReduxState): AdminHomeMappedProps => ({
  adminState: state.admin
})

export default connect(mapStateToProps)(AdminHome)
