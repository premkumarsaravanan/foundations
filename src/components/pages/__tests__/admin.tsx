import * as React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { AdminHome, AdminHomeProps } from '../admin'

const props: AdminHomeProps = {
  adminState: {
    loading: false,
    appRevisions: []
  }
}

describe('AdminHome', () => {
  it('should match a snapshot', () => {
    expect(toJson(shallow(<AdminHome {...props} />))).toMatchSnapshot()
  })
})
