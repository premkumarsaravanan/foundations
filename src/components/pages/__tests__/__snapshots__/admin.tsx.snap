// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminHome should match a snapshot 1`] = `
<div
  className="container"
>
  <Connect(ErrorBoundary)>
    <div>
      Admin area
    </div>
  </Connect(ErrorBoundary)>
</div>
`;
